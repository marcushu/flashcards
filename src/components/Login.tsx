import { useState } from 'react';

interface LoginProps {
  login: (input: string) => void
  signUp: (intput: string) => void
}

const Login = ({login, signUp}: LoginProps) => {
  const [userName, setuserName] = useState("");
  const [newUserName, setnewUserName] = useState("");

  return (
    <div className="card">
      <div>
        <input className="loginText"
          type="text"
          alt="login"
          onChange={e => setuserName(e.currentTarget.value)} />
        <br /><br />
        <button className="loginBtn" onClick={() => login(userName)}>
          Log In
            </button>
        <h4>Or</h4>
        <input className="loginText"
          type="text"
          alt="signup"
          onChange={e => setnewUserName(e.currentTarget.value)} />
        <br /><br />
        <button className="loginBtn" onClick={() => signUp(newUserName)}>
          Sign up
        </button>
      </div>
    </div>
  )
}

export default Login;