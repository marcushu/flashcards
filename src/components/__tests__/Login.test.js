import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Login from '../Login';

describe("Login", () => {
    const mockLogin = jest.fn();
    const mockSignup = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <Login login={mockLogin} signUp={mockSignup} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("calls login with correct name", () => {
        const { getByAltText, getByRole } = render(
            <Login login={mockLogin} signUp={mockSignup} />
        );

        fireEvent.change(getByAltText(/login/i), { target: {value: "myname"}});
        
        fireEvent.click(getByRole('button', {  name: /log in/i}));

        expect(mockLogin).toHaveBeenCalledWith("myname");
    });

    it("calls sign up with correct name", () => {
        const { getByAltText, getByRole } = render(
            <Login login={mockLogin} signUp={mockSignup} />
        );

        fireEvent.change(getByAltText(/signup/i), { target: {value: "newname"}});
        
        fireEvent.click(getByRole('button', {  name: /sign up/i}));

        expect(mockSignup).toHaveBeenCalledWith("newname");
    });
})