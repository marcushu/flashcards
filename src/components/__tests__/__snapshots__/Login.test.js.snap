// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div>
      <input
        alt="login"
        class="loginText"
        type="text"
      />
      <br />
      <br />
      <button
        class="loginBtn"
      >
        Log In
      </button>
      <h4>
        Or
      </h4>
      <input
        alt="signup"
        class="loginText"
        type="text"
      />
      <br />
      <br />
      <button
        class="loginBtn"
      >
        Sign up
      </button>
    </div>
  </div>
</DocumentFragment>
`;
